### PR TITLE
nrf_security: fix random number gen on nrf52833

### PR DESCRIPTION
**Pull request by MarekPorwisz**
mbedtls_hardware_poll was not doing anything but is needed on nrf52833.
Fixed the function for proper random numbers generation.

manifest pr: https://github.com/nrfconnect/sdk-nrf/pull/3025

Signed-off-by: Marek Porwisz <marek.porwisz@nordicsemi.no>

### CODE REVIEW

File: nrf_security/src/backend/nrf5x/entropy_nrf5x.c [hunk 1]
```diff
@@ -15,42 +15,42 @@ int mbedtls_hardware_poll(void *data,
...
-    {
+    if (!dev) {
         return MBEDTLS_ERR_ENTROPY_NO_SOURCES_DEFINED;
     }
 
-    ret = entropy_get_entropy(dev, buffer, MBEDTLS_ENTROPY_MAX_GATHER);
-    if (ret)
-    {
-        return MBEDTLS_ERR_ENTROPY_SOURCE_FAILED;
+    while ( len >0 ) {
```
Comment (edmont) [inline nrf_security/src/backend/nrf5x/entropy_nrf5x.c:44]:
```suggestion
    while ( len > 0 ) {
```

File: nrf_security/src/backend/nrf5x/entropy_nrf5x.c [hunk 2]
```diff
@@ -15,42 +15,42 @@ int mbedtls_hardware_poll(void *data,
...
+        chunk_size = MIN(MBEDTLS_ENTROPY_MAX_GATHER, len);
+
+        if (entropy_get_entropy(dev, output, chunk_size) <0) {
```
Comment (edmont) [inline nrf_security/src/backend/nrf5x/entropy_nrf5x.c:47]:
```suggestion
        if (entropy_get_entropy(dev, output, chunk_size) < 0) {
```

**Review by edmont — APPROVED** [review]
This looks very good to me, it fixes both KRKNWK-7652 and KRKNWK-7656.

Just some nits in the comments. Also please add a new line before the includes at the beginning of the file.

**Comment by rlubos** [comment]
@tejlmand @frkv What coding standard shall this file follow, mbedTLS, NCS?

**Comment by frkv** [comment]
> @tejlmand @frkv What coding standard shall this file follow, mbedTLS, NCS?

The code is following mbed TLS coding standard

File: nrf_security/src/backend/nrf5x/entropy_nrf5x.c [hunk 3]
```diff
@@ -35,22 +34,25 @@ int mbedtls_hardware_poll(void *data,
...
     dev = device_get_binding(DT_CHOSEN_ZEPHYR_ENTROPY_LABEL);
 
     if (!dev)
     {
         return MBEDTLS_ERR_ENTROPY_NO_SOURCES_DEFINED;
     }
 
-    ret = entropy_get_entropy(dev, buffer, MBEDTLS_ENTROPY_MAX_GATHER);
-    if (ret)
+    while ( len >0 )
```
Comment (edmont) [inline nrf_security/src/backend/nrf5x/entropy_nrf5x.c:44]: I would place a spaces between the two operands and the operator here.

**Review by frkv — APPROVED** [review]
LGTM

**Comment by tejlmand** [comment]
@MarekPieta where is the manifest PR ?

**Comment by MarekPorwisz** [comment]
> @MarekPorwisz where is the manifest PR ?

https://github.com/nrfconnect/sdk-nrf/pull/3025

**Review by tejlmand — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Pull request merged**